packages/firestore/src/remote/bloom_filter.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Md5, Integer } from '@firebase/webchannel-wrapper';
+
+import { newTextEncoder } from '../platform/serializer';
+import { debugAssert } from '../util/assert';
+
+const MAX_64_BIT_UNSIGNED_INTEGER = Integer.fromNumber(Math.pow(2, 64));
+export class BloomFilter {
+  private readonly size: Integer;
```

Comment:
I think BloomFilter should store the size in two variables: a plain-old `number` and a fancy `Integer`. The "plain old number" should be public and the fancy Integer should be private. It is expensive to convert between these two types so I think it makes sense to perform the computation once, in the constructor, and assign to those instance variables.